### ISSUE TITLE
Add Support for Conditional Access Expressions

### ISSUE THREAD

**Issue by j0nimost** [MapsterMapper/Mapster#708]
Currently, we cannot use conditional access expressions while declaring mappings. For example;

```c#
TypeAdapterConfig<ItemA, ItemB>.NewConfig()
    .Map(dest => dest.Name, src => src.ItemAChild?.Name);
```
The above example would fail with an error, however, conditional access is very common when mapping values. This is a needed feature.

**Comment by DocSvartz** [comment]
Hello @j0nimost. Can you explain what behavior you want to get?

`src => src.ItemAChild?.Name` this construction should interrupt the update if the property in the source `ItemAChild == null`   or other behavior ?

[There is already a mechanism for matching with a condition](https://github.com/MapsterMapper/Mapster/wiki/Custom-mapping#mapping-with-condition)

**Comment by j0nimost** [comment]
Yes, ideally if `ItemAChild == null` then the default value should be set for `dest.Name` else map `src.ItemAChild.Name` to destination. From the link shared I can see how 
```.Map(dest => dest.Name, src => src.ItemAChild.Name, srcCond => srcCond.ItemAChild != null)``` 
fixes this. I hadn't seen this I assumed Null-conditional operators in C# would work out of the box

**Comment by DocSvartz** [comment]
As far as I understand. This is all an instruction (not real code) for mapping- "Custom Name Mapping" for different Types.

It seems that this is why a two-stage design was chosen:
1) What we will match in two Types.
2) How we will do it (Conditions or restrictions).

Therefore, using a conditional access expression at stage 1 does not make sense. It simply defines the source and destination for mapping (Property names in Types)..

I understand what you want to get, but so far I have no ideas on how to do this.

**Comment by j0nimost** [comment]
What triggered this query was the compiler warnings I was getting. The `null-suppression` didn't cut it for me as a solution, especially for the `Map(source, destination)` . I am now well aware of the `Map(source, destination, condition)` 3rd parameter that can be used.

**Comment by andrerav** [comment]
Sounds like this has been solved then :) If you need more assistance with usage, feel free to open a discussion thread. Thank you very much for assisting @DocSvartz!
